### PR TITLE
utils: reorg

### PR DESCRIPTION
**Pull request by emersonmacro**
Took a stab at issue https://github.com/NebulousLabs/skynet-js/issues/183

### CODE REVIEW

**Comment by mrcnski** [comment]
Thank you for your contribution! There may be a delay in getting this reviewed, as we have a very busy period right now. We'll get to it as soon as possible. The overall structure of the PR is a 👍 from me

File: src/utils/skylink.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+// TODO: This will be smarter. See
+// https://github.com/NebulousLabs/skynet-docs/issues/21.
+/**
+ * Returns the default portal URL.
+ *
+ * @returns - The portal URL.
+ */
+export function defaultPortalUrl(): string {
```
Comment (MSevey) [inline src/utils/skylink.ts]: I would put this in the url file.

File: src/utils/utils.test.ts [hunk 2]
```diff
@@ -1,22 +1,16 @@
+import { getFileMimeType, getRelativeFilePath, getRootDirectory } from "./file";
```
Comment (MSevey) [inline src/utils/utils.test.ts]: We should also split this test file out so we have `url.ts` and `url.test.ts` for example.

**Review by MSevey — CHANGES_REQUESTED** [review]

File: src/utils/skylink.ts (hunk 1 above)
Comment (emersonmacro) [reply src/utils/skylink.ts]: Good call. I wasn't sure where that one should go. PR updated.

File: src/utils/utils.test.ts (hunk 2 above)
Comment (emersonmacro) [reply src/utils/utils.test.ts]: Split out tests for each file 👍

**Comment by emersonmacro** [comment]
Not sure why some of the CI builds failed but the 15.x build was successful... `yarn test` passed all tests for me locally, running node 12.x. Any suggestions?

**Review by MSevey — APPROVED** [review]

**Comment by MSevey** [comment]
> Not sure why some of the CI builds failed but the 15.x build was successful... `yarn test` passed all tests for me locally, running node 12.x. Any suggestions?

Going to have to look to Marcin on that one.

File: src/utils/skylink.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import parse from "url-parse";
+import { CustomClientOptions } from "../client";
+import { trimForwardSlash, trimSuffix, trimUriPrefix } from "./string";
+
+/**
+ * Base custom options for methods hitting the API.
+ *
+ * @property [endpointPath] - The relative URL path of the portal endpoint to contact.
+ */
+export type BaseCustomOptions = CustomClientOptions & {
```
Comment (mrcnski) [inline src/utils/skylink.ts:12]: The options stuff like `BaseCustomOptions` and `defaultOptions`. doesn't seem to fit in `skylink.ts`. Maybe `utils/options.ts`? `ParseSkylinkOptions` should stay here.

**Review by mrcnski — APPROVED** [review]
I had one small comment, but it's not blocking. The PR looks good as is. Thank you!

**Pull request merged**